### ISSUE TITLE
JS hash changes when src folder moved

### ISSUE THREAD

**Issue by jakearchibald** [rollup/rollup#3416]
- Rollup Version: 1.32.0
- Operating System (or Browser): OSX
- Node Version (if applicable): v12.16.1
- Link to reproduction: https://github.com/jakearchibald/rollup-hash-bug-2

### Expected Behavior

The two `src` folders have identical contents, their output filenames should not differ between builds.

### Actual Behavior

All three files change hash.

#2839 may be related.

**Comment by lukastaegert** [comment]
Interesting point. My feeling is that implementing the improved hashing I  outlined in https://github.com/rollup/rollup/issues/2872#issuecomment-591846400 might solve this as well, definitely worth keeping as a test case for a new algorithm.

**Comment by jakub-g** [comment]
I don't know if related, but I was hitting the same kind of issue with webpack, and the problem was that a loader (plugin) transforming an import request used _absolute disk path_ of the file as an internal import identifier (relying on `__dirname`, `path.resolve()` and other similar things can lead to this), leading to non-deterministic input (when building inside a different folder for example).

**Comment by myarcane** [comment]
It is worth mentioning that the bug only exists with **code splitting** configurations when `input` takes an array of files :
```javascript
 input: [`src${num}/index.js`, `src${num}/profile.js`]
 ```
 Obviously if rollup builds each of these files separately : 
 ```javascript
 // rolllup.config.js
 export default ["index.js", "profile.js"].reduce((config, file) => {
  [1, 2].forEach((num) => {
    config.push({
      input: `src${num}/${file}`,
      output: {
        entryFileNames: "[name]-[hash].js",
        dir: "build/" + num,
        format: "esm",
      },
    });
  });
  return config;
}, []);
```
the hash names remain the same for each source folder.

Maybe a "code splitting" label should be added ?

**Comment by lukastaegert** [comment]
Fix finally at #4543

**Comment by lukastaegert** [comment]
Fix finally at #4543
